lang/ksp/src/main/kotlin/nameModels.kt

```diff
@@ -50,14 +51,18 @@ private fun nameModelImpl(
         JvmTypeInfo.Short -> KeywordTypeNameModel.Short
         JvmTypeInfo.Boolean -> KeywordTypeNameModel.Boolean
         JvmTypeInfo.Declared -> {
-            checkNotNull(type) { "Not reached: type info is absent for declared type" }
-            if (type.isError) {
-                return ErrorNameModel()
+            if (type == null || type.isError) {
```

Comment:
Does isError means no info at all here?

Reply:
Yes, sadly. In current KSP implementation the the error-type is in fact an `object` that just does `"<ERROR TYPE>"` in it's `toString()`. There's an issue for KSP, linked in #24. For now all we can do is try to recover a simple (unqualified) name for a type from its source code reference - it's done in `KspTypeImpl.obtainType()`.

Reply:
There was actually a chance to salvage some info about unresolved type arguments, as they are also presented as references whose `element`s might contain some text. But alas, KSP looses all the info along the way. Consider `List<UnresolvedEp>`, where `List` is a normal type and `UnresolvedEp` is an unresolved type.  Then, `type.toString()` is `List<[Error type: Unresolved type for UnresolvedEp]>` while `type.arguments.toString()` is already `[INVARIANT <Error class: unknown class>]`, and `element` yields `<Error class: unknown class>` respectively. So, KSP does a great job of loosing the last drop of info that could help us to identify the unresolved type. I'm not desperate enough to parse `type.toString()` in search of type arguments, at least for now..